lib/montage/client.rb

```diff
@@ -13,32 +13,82 @@ class Client
     include Schemas
     include Documents
 
-    attr_accessor :token, :username, :password, :domain, :api_version, :url_prefix
-
+    attr_accessor :token,
+                  :username,
+                  :password,
+                  :domain,
+                  :api_version,
+                  :environment
+
+    # Initializes the client instance
+    #
+    # * *Attributes*    :
+    #   - +token+ -> API access token required for requests, does not expire
+    #   - +username+ -> Montage username credential
+    #   - +password+ -> Montage username credential
+    #   - +domain+ -> Project subdomain, required for initiliaztion
+    #   - +api_version+ -> API version to query against, defaults to 1
+    #   - +environment+ -> Specifies desired environment for requests, defaults to 'production'
+    # * *Returns* :
+    #   - A valid Montage::Client instance, you hope
+    # * *Raises* :
+    #   - +MissingAttributeError+ -> If the domain attribute is not specified
+    #   - +InvalidEnvironment+ -> If the environment attribute is not set to 'production' or 'development'
+    #
     def initialize
       @api_version = 1
+      @environment = 'production' if @environment.nil?
       yield(self) if block_given?
-      raise MissingAttributeError, "You must declare the domain attribute" unless @domain
+      fail MissingAttributeError, 'You must declare the domain attribute' unless @domain
+      fail InvalidEnvironment, "Valid options are 'production' and 'development'" unless environment_valid?
     end
 
-    def default_url_prefix
-      "https://#{domain}.mntge.com"
+    # Verifies the Montage::Client instance environment
+    #
+    # * *Returns* :
+    #   - A boolean
+    #
+    def environment_valid?
+      @environment == 'production' || @environment == 'development' ? true : false
     end
 
-    def content_type
-      "application/json"
+    # Generates a base url for requests using the supplied environment and domain
+    #
+    # * *Returns* :
+    #   - A string containing the constructed url
+    #
+    def default_url_prefix
+      return "https://#{domain}.dev.montagehot.club" if @environment == 'development'
+      return "https://#{domain}.mntge.com" if @environment == 'production'
     end
 
+    # Attempts to authenticate with the Montage API
+    #
+    # * *Returns* :
+    #   - A hash containing a valid token or an error string, oh no!
+    #
     def auth
-      build_response("token") do
+      build_response('token') do
```

Comment:
Not sure if your editor is telling you to replace the double quotes, but stylistically, I prefer double quotes everywhere. Just to remain consistent. The speed difference is so ridiculously minimal that it's not in any way relevant for this gem.